src/iuse.cpp

```diff
@@ -7490,48 +7490,80 @@ int iuse::ladder( player *p, item *, bool, const tripoint& )
     return 1;
 }
 
+
 int iuse::washclothes( player *p, item *it, bool, const tripoint& )
 {
+    // Check that player isn't over volume limit as this might cause it to break... this is a hack.
+    // TODO: find a better solution.
+    if (p->volume_capacity() < p->volume_carried()){
+        p->add_msg_if_player( _( "You're carrying too much to clean anything." ) );
+        return 0;
+    }
     if( it->charges < it->type->charges_to_use() ) {
         p->add_msg_if_player( _( "You need a cleansing agent to use this." ) );
         return 0;
     }
 
-    const int pos = g->inv_for_flag( "FILTHY", _( "Wash what?" ) );
-    item &mod = p->i_at( pos );
-    if( pos == INT_MIN ) {
-        p->add_msg_if_player( m_info, _( "Never mind." ) );
+
+
+    player player = *p;
+    
+    player.inv.restack( player );
+
+    const inventory_filter_preset preset( []( const item_location & location ) {
+        return (*location).item_tags.find("FILTHY") != (*location).item_tags.end();
```

Comment:
Instead of calling `operator*` (which requires the parentheses in this context), you can call `operator->`: `location->item_tags`.

Same with `(*pairs).second` below.